src/generators/AbstractRenderer.ts

```diff
@@ -11,8 +11,21 @@ export abstract class AbstractRenderer<O extends CommonGeneratorOptions = Common
     protected readonly presets: Array<[Preset, unknown]>,
     protected readonly model: CommonModel, 
     protected readonly inputModel: CommonInputModel,
+    public dependencies: string[] = []
   ) {}
 
+  /**
+   * Adds a dependency while ensuring that only one dependency is preset at a time.
+   * 
+   * @param dependency complete dependency string so it can be rendered as is.
+   */
+  addDependency(dependency: string): void {
+    this.dependencies = this.dependencies || [];
```

Comment:
```suggestion
```

If you add as default value in the constructor for `dependencies` the empty array, you shouldn't reassign it here :) I know why you did this, but it's unnecessary.